pydantic/env_settings.py

```diff
@@ -134,30 +136,47 @@ def __repr__(self) -> str:
 class EnvSettingsSource:
```

Comment:
I wonder if it would be smarter to split dotenv processing into a separate source?

I suggested the same thing [here](https://github.com/samuelcolvin/pydantic/pull/3159#discussion_r765249632) but this case is more clear cut.

Reply:
Could you elaborate on it?

Reply:
See how sources work in settings. I propose moving dotenv logic into a whole new class.

Reply:
Good idea, but it will break backwards compatibility. `custom_sources` should be changed to take 4 callables as arguments. Any ideas?

Reply:
let's leave to v2.